sdk/core/core-client-rest/src/sendRequest.ts

```diff
@@ -37,11 +37,42 @@ export async function sendRequest(
 ): Promise<HttpResponse> {
   const httpClient = customHttpClient ?? getCachedDefaultHttpsClient();
   const request = buildPipelineRequest(method, url, options);
+  if (options) {
+    if (options.timeout) {
```

Comment:
if `options.timeout` is set to `0` this will be falsy and won't set the value onto `request.timeout`

Since presumably `request.timeout` can be undefined, are we able to avoid the if checks for values here and always do the assignment?

```ts
request.timeout = options.timeout;
request.onUploadProgess = options.onUploadProgress;
// etc
```